### PR TITLE
feat(CTA-6) Delete Appointment

### PR DESCRIPTION
**Pull request by drallo22**
JIRA: https://champlainsaintlambert.atlassian.net/jira/software/projects/CTA/boards/59/backlog?selectedIssue=CTA-6
## Context:
Implemented the functionality for a user (client/admin) to delete/cancel an appointment
## Changes

- Implemented backend 
- Implemented frontend
- Implemented backend and frontend testing
- Implemented validation in frontend that does not allow a user to cancel an appointment the day of or before their appointment
- Made fixes in the way an appointment and quote request was scheduled

##  UI (Required for UI-impacting PRs)
image

### CODE REVIEW

**Review by Matthewtan9 — APPROVED** [review]
LGTM

File: coupetonarbrefrontend/src/pages/AppointmentsClient.jsx [hunk 1]
```diff
@@ -111,6 +113,41 @@ const formatDate = (dateString) => {
...
+      // Current year
+      const currentyear = moment(new Date()).year();
+  
+      // Appointment day and year
+      const appointmentDay = moment(appointmentDetails.date).dayOfYear();
+      const appointmentYear = moment(appointmentDetails.date).year();
+  
+      // Check if the appointment is on the day of or the next day
+      if (currentyear === appointmentYear && (appointmentDay === currentDay + 1 || appointmentDay === currentDay)) {
+        alert("Cannot cancel an appointment the day before or of!");
```
Comment (dazhuox) [inline coupetonarbrefrontend/src/pages/AppointmentsClient.jsx:129]: The sentence is a bit weird here, I think "of" comes before "before"? I'm not sure but the sentence seems a bit off

**Review by dazhuox — APPROVED** [review]
Great job, looks good and seems like everything is consistent with quoterequests

File: coupetonarbrefrontend/src/pages/AppointmentsClient.jsx (hunk 1 above)
Comment (drallo22) [reply coupetonarbrefrontend/src/pages/AppointmentsClient.jsx]: Yeah I agree, I changed it to make it a bit clear thanks!

Comment (dazhuox) [reply coupetonarbrefrontend/src/pages/AppointmentsClient.jsx]: Thanks!!

**Review by dazhuox — APPROVED** [review]
Looks good! Seems consistent with quoteRequest, great job.

**Pull request merged**